spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -79,9 +81,19 @@ public GcpPubSubAutoConfiguration(GcpPubSubProperties gcpPubSubProperties,
 		this.finalProjectIdProvider = gcpPubSubProperties.getProjectId() != null
 				? gcpPubSubProperties::getProjectId
 				: gcpProjectIdProvider;
-		this.finalCredentialsProvider = gcpPubSubProperties.getCredentials().hasKey()
-				? new DefaultCredentialsProvider(gcpPubSubProperties)
-				: credentialsProvider;
+
+		if (StringUtils.isEmpty(gcpPubSubProperties.getEmulatorHost())
+				|| "false".equals(gcpPubSubProperties.getEmulatorHost())) {
```

Comment:
What does this second condition test? How can the `emulatorHost` property be `false`?

Reply:
`@ConditionalOnProperty(prefix = "spring.cloud.gcp.pubsub", name = "emulator-host")` matches also if I set `spring.cloud.gcp.pubsub.emulator-host=` (to empty)

Reply:
Apologies if I'm missing something, but isn't that what the first condition tests?

Reply:
yes, if I set `spring.cloud.gcp.pubsub.emulator-host=`
* `@ConditionalOnProperty(prefix = "spring.cloud.gcp.pubsub", name = "emulator-host")` will match and
* `StringUtils.isEmpty(gcpPubSubProperties.getEmulatorHost())` will match

if I set `spring.cloud.gcp.pubsub.emulator-host=false`
* `@ConditionalOnProperty(prefix = "spring.cloud.gcp.pubsub", name = "emulator-host")` will not match and
* `"false".equals(gcpPubSubProperties.getEmulatorHost())` will match

So a simple `null` check and the test to `flase` will match `@ConditionalOnProperty(prefix = "spring.cloud.gcp.pubsub", name = "emulator-host")` better, I will adjust this.